src/module/config.h

```diff
@@ -17,3 +18,23 @@ extern unsigned int krf_rng_state;
 extern unsigned int krf_personality;
 extern unsigned int krf_probability;
 extern unsigned int krf_log_faults;
+extern unsigned int krf_targeting;
+
+#define KRF_T_MODE_MAX 31
+#define KRF_T_MODE_MAX_MASK (1 << KRF_T_MODE_MAX)
+
+typedef enum {
+  KRF_T_MODE_PERSONALITY = 0,
+  KRF_T_MODE_PID,
+  KRF_T_MODE_UID,
+  KRF_T_MODE_GID,
+  // Insert new modes here
+  KRF_T_NUM_MODES
+} krf_target_mode_t;
```

Comment:
Nit: Add a `static_assert` or `#error` here to compile-time assert that `KRF_T_NUM_MODES <= KRF_T_MODE_MAX`.